### PR TITLE
RPCDD4HEP bis

### PR DESCRIPTION
**Pull request by slomeo**
#### PR description:

DD4HEP migration for RPC

Test: see pdf files 
[Canvas_1.pdf](https://github.com/cms-sw/cmssw/files/3635177/Canvas_1.pdf)
[Canvas_2.pdf](https://github.com/cms-sw/cmssw/files/3635178/Canvas_2.pdf)
[Canvas_3.pdf](https://github.com/cms-sw/cmssw/files/3635179/Canvas_3.pdf)
[Canvas_4.pdf](https://github.com/cms-sw/cmssw/files/3635181/Canvas_4.pdf)
[Canvas_5.pdf](https://github.com/cms-sw/cmssw/files/3635182/Canvas_5.pdf)
[Canvas_6.pdf](https://github.com/cms-sw/cmssw/files/3635183/Canvas_6.pdf)
[Canvas_7.pdf](https://github.com/cms-sw/cmssw/files/3635184/Canvas_7.pdf)
[Canvas_8.pdf](https://github.com/cms-sw/cmssw/files/3635185/Canvas_8.pdf)

made by validation process

### CODE REVIEW

**Comment by slomeo** [comment]
I have run "scram build code-checks" and "scram build code-format"

File: Geometry/MuonNumbering/src/RPCNumberingScheme.cc [hunk 1]
```diff
@@ -34,6 +42,13 @@ void RPCNumberingScheme::initMe(const MuonDDDConstants& muonConstants) {
 }
 
 int RPCNumberingScheme::baseNumberToUnitNumber(const MuonBaseNumber& num) {
+  const int mylevel = num.getLevels();
+  //cout<<"------------myDEBUG DDD-------------------------------------------"<<endl;
```
Comment (fabiocos) [inline Geometry/MuonNumbering/src/RPCNumberingScheme.cc]: @slomeo please avoid commented printout and use the MessageLogger instead (possibly with pre-processor conditions)

**Comment by fabiocos** [comment]
please test

File: Geometry/MuonNumbering/src/RPCNumberingScheme.cc (hunk 1 above)
Comment (slomeo) [reply Geometry/MuonNumbering/src/RPCNumberingScheme.cc]: @fabiocos I deleted all not necessary "cout" and "include"

> @slomeo please avoid commented printout and use the MessageLogger instead (possibly with pre-processor conditions)

**Comment by fabiocos** [comment]
please test

**Comment by slomeo** [comment]
I made a test by using workflows 21.0 and 4.53. It seems all ok.

**Comment by fabiocos** [comment]
please test

File: Geometry/MuonNumbering/interface/DD4hep_RPCNumberingScheme.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class MuonBaseNumber;
+class MuonConstants;
+
+namespace cms {
+  class RPCNumberingScheme {
+  public:
+    RPCNumberingScheme(const MuonConstants& muonConstants);
+
+    void baseNumberToUnitNumber(const MuonBaseNumber&);
+    void SetDetId(int idnew) { detId = idnew; }
```
Comment (ianna) [inline Geometry/MuonNumbering/interface/DD4hep_RPCNumberingScheme.h]: please, use CMS naming conventions: a member function/method name starts with a small letter.
BTW, this method should be private.

File: Geometry/MuonNumbering/interface/DD4hep_RPCNumberingScheme.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    int GetDetId() const { return detId; }
```
Comment (ianna) [inline Geometry/MuonNumbering/interface/DD4hep_RPCNumberingScheme.h]: please, follow the CMS naming conventions

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+      } else {
+        copy_id = 1;
+      }
+    }
+    if (barrel_muon) {
+      if (level == theBWheelLevel) {
+        const int copyno = num.getBaseNo(level);
+        eta_id = 4 + copyno;  //copyno= [0,4]
+      } else if (level == theBStationLevel) {
+        //-	const int station_tag = num.getSuperNo(level);
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        } else if (sector_id == 14) {
+          sector_id = 10;
+          sector_copy = 1;
+        }
+        // mltiply by 3 to merge with endcaps
+        sector_id *= 3;
+
+      } else if (level == theBPlaneLevel) {
+        const int plane_tag = num.getSuperNo(level);
+        //        const int copyno = num.getBaseNo(level);
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+        if (plane_tag == 1) {
+          plane_id = 1;
+        } else if (plane_tag == 2) {
+          plane_id = 5;
+        } else if (plane_tag == 3) {
+          //          if(copyno == 1) {
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove commented out code

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+          //if(eta_id == 4 || eta_id == 8) {
+          //  plane_id=6;
+
+          plane_id = 2;
+          // }
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove commented out code

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+
+          //          std::cout<<" KONTROLA w RPCNumberingScheme: eta_id: "<<eta_id<<", plane_tag: "<<plane_tag<<", plane_id: "<<plane_id<<std::endl;
+        } else if (plane_tag == 4) {
+          //          if(copyno == 1) {
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove commented out code

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+          // if(eta_id == 4 || eta_id == 8) {
+          //  plane_id=2;
+          //} else {
+          plane_id = 6;
+          //}
```
Comment (ianna) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: please, remove commented out code

File: Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.cc [hunk 10]
```diff
@@ -25,14 +32,21 @@
```
Comment (ianna) [inline Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.cc]: please, remove CLHEP and use only CMSUnits

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.cc (hunk 10 above)
Comment (slomeo) [reply Geometry/RPCGeometryBuilder/src/RPCGeometryBuilderFromDDD.cc]: @ianna Done!

**Comment by ianna** [comment]
please test

**Comment by ianna** [comment]
+1

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+
+  bool barrel_muon = (barrel == 1);
+  int maxLevel;
+  if (barrel_muon) {
+    maxLevel = theBChamberLevel;
+  } else {
+    maxLevel = theERollLevel;
+  }
+  if (num.getLevels() != maxLevel) {
+    abort();
```
Comment (fabiocos) [inline Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: @slomeo @ianna why not throwing a cms::Exception?

File: Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc [hunk 12]
```diff
@@ -0,0 +1,305 @@
+/*
+//\class RPCGeometryValidate
+
```
Comment (fabiocos) [inline Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc]: @slomeo in which context is this supposed to be used? Not the DQM workflows, just for standalone validation?

Comment (slomeo) [reply Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc]: @fabiocos It is used only for standalone validation (i.e histos production)

File: Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc (hunk 11 above)
Comment (slomeo) [reply Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: @fabiocos it was a  line code I used during debug...I'm sorry.. do I have to change it (and commit again)?

Comment (slomeo) [reply Geometry/MuonNumbering/src/DD4hep_RPCNumberingScheme.cc]: @fabiocos fixed

File: Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc (hunk 12 above)
Comment (fabiocos) [reply Geometry/RPCGeometryBuilder/plugins/RPCGeometryValidate.cc]: @slomeo so it would be better to have it in a test directory

**Comment by fabiocos** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**